### PR TITLE
Implement a Flag on Model for if the model should be called with units

### PR DESCRIPTION
**Pull request by Cadair**
This adds a property to Model that allows introspection of if the Model allows Quantity objects to it's `__call__`.

### CODE REVIEW

File: astropy/modeling/core.py [hunk 1]
```diff
@@ -728,6 +728,21 @@ def input_units_allow_dimensionless(self):
...
+        objects or if there are no parameters.
+
+        This can be used to determine if this model should be evaluated with
+        `~astropy.units.Quantity` or regular floats.
+        """
+        pisq = [isinstance(p, Quantity) for p in self._param_sets(units=True)]
+        if not len(pisq) or any(pisq):
+            return True
+        else:
+            return False
```
[the remark this answers is not in the document]
Comment (saimn) [reply astropy/modeling/core.py]: Right, I was confused by the `not len` ;)
I still think you can avoid the if/else though, what about `return (len(pisq) == 0) or any(pisq)` ?

**Comment by pllim** [comment]
@nden , yes 3.0.3 is only for bug fix. API change or new feature go to 3.1.

**Comment by pllim** [comment]
(I remilestoned it because it is not marked as a bug. Feel free to change if that is not the case.)

**Comment by Cadair** [comment]
@nden updated.

**Pull request merged**